### ISSUE TITLE
GP training documentation

### ISSUE THREAD

**Issue by cherepanovic** [cornellius-gp/gpytorch#815]
Are there documentations about the training process of GP? 

thanks!

**Comment by cherepanovic** [comment]
I found this:
![train](https://user-images.githubusercontent.com/10064548/62141685-4b30e680-b2ed-11e9-8cf7-249e0262365b.png)

from third section of https://arxiv.org/pdf/1809.11165.pdf paper, it is the whole magic behind the training?

**Comment by cherepanovic** [comment]
that means, that a model with a RBF kernel and a Gaussian likelihood would have only following parameters: lengthscale (RBF), mean and the parameters of Gaussian likelihood (noise, sigma).

Learning happens by means of an optimizer such as Adam with a learning rate, as can be seen from code, right?

**Comment by gpleiss** [comment]
Hi @cherepanovic - training happens in the optimization loop (with an optimizer such as Adam), such as the optimization loop in [this tutorial](https://gpytorch.readthedocs.io/en/latest/examples/01_Simple_GP_Regression/Simple_GP_Regression.html).

In general, you typically only have the hyperparameters that you discuss - though if you have a more complex kernel, you'll have more hyperparameters. Saving the model.state_dict() will save these hyperparametrs. It is not necessary to save the covariance matrix.

If you want to access the model's prior training-data covariance matrix, set `model.train()`, and then call `model(train_x).covariance_matrix`.

**Comment by cherepanovic** [comment]
@gpleiss 
> It is not necessary to save the covariance matrix.

after having trained a model, then I would save it and reload at a later point and apply on some test data. For that a GP must have the covariance of the training data in order to calculate the prediction for the test data. How it is dealt with (without saved covariance of the training data)? 

Thank a lot!

**Comment by cherepanovic** [comment]
@gpleiss 

I am still struggling with understanding regarding K_xx, why it will not be saved... How does the prediction (for the test data) work without it?

Thanks a lot!

**Comment by gpleiss** [comment]
I'm going to close this thread in favor of the one in #902
